src/NzbDrone.Core/Datastore/Migration/023_add_release_groups_etc.cs

```diff
@@ -0,0 +1,246 @@
+                .WithColumn("Media").AsString().Nullable()
+                .WithColumn("TrackCount").AsInt32().Nullable()
+                .WithColumn("Monitored").AsBoolean();
+
+            Execute.WithConnection(PopulateReleases);
+
+            // ALBUMS TABLE
+
+            // Add in the extra columns and update artist metadata id
+            Alter.Table("Albums").AddColumn("ArtistMetadataId").AsInt32().WithDefaultValue(0);
+            Alter.Table("Albums").AddColumn("AnyReleaseOk").AsBoolean().WithDefaultValue(true);
+
+            // Set metadata ID
+            Execute.Sql(@"UPDATE Albums
+                          SET ArtistMetadataId = (SELECT ArtistMetadata.Id 
+                                                  FROM ArtistMetadata 
+                                                  JOIN Artists ON ArtistMetadata.Id = Artists.ArtistMetadataId
+                                                  WHERE Albums.ArtistId = Artists.Id)");
+            
+            // TRACKS TABLE
+            Alter.Table("Tracks").AddColumn("ForeignRecordingId").AsString().WithDefaultValue("0");
+            Alter.Table("Tracks").AddColumn("AlbumReleaseId").AsInt32().WithDefaultValue(0);
+            Alter.Table("Tracks").AddColumn("ArtistMetadataId").AsInt32().WithDefaultValue(0);
+            
+            // Set track release to the only release we've bothered populating
+            Execute.Sql(@"UPDATE Tracks
+                          SET AlbumReleaseId = (SELECT AlbumReleases.Id 
+                                                FROM AlbumReleases
+                                                JOIN Albums ON AlbumReleases.AlbumId = Albums.Id
+                                                WHERE Albums.Id = Tracks.AlbumId)");
+
+            // CLEAR OUT OLD COLUMNS
+
+            // Remove the columns in Artists now in ArtistMetadata
+            Delete.Column("ForeignArtistId")
+                .Column("Name")
+                .Column("Overview")
+                .Column("Disambiguation")
+                .Column("ArtistType")
+                .Column("Status")
+                .Column("Images")
+                .Column("Links")
+                .Column("Genres")
+                .Column("Ratings")
+                .Column("Members")
+                // as well as the ones no longer used
+                .Column("MBId")
+                .Column("AMId")
+                .Column("TADBId")
+                .Column("DiscogsId")
+                .Column("NameSlug")
+                .Column("LastDiskSync")
+                .Column("DateFormed")
+                .FromTable("Artists");
+
+            // Remove old columns from Albums
+            Delete.Column("ArtistId")
+                .Column("MBId")
+                .Column("AMId")
+                .Column("TADBId")
+                .Column("DiscogsId")
+                .Column("TitleSlug")
+                .Column("Label")
+                .Column("SortTitle")
+                .Column("Tags")
```

Comment:
My view is that stuff that isn't used/properly tested should be removed and can always be put back in when the album-centric actually happens.  Otherwise it might accidentally get used when it doesn't work properly.  But happy to leave it if you guys prefer.

Reply:
I’m good adding back when used.